### DISCUSSION TITLE
Configmap for nginx ingress controller not taking effect

### DISCUSSION THREAD

**Discussion by rrachitha** [nginx/kubernetes-ingress#2816] [Q&A]
Issue:

We upgraded our AKS cluster from v1.19.x to v1.22.6 and post upgrade we had to update the Nginx-ingress docker image to use version 2.2 which is nginx version: 1.21.6 after this change the changes in the configmap configurations are getting ignored and not taking any effect and most of the default configurations are being applied for now. 

We looked into the logs for nginx pods as well as the events for configmap post update, no errors were found also no events related to this.

Note: Nginx ingress controller is managed by a deployment definition file.

**First Question:**
Even though I see the docker image's command starts with `/nginx-ingress` as per the below screenshot I still need to pass the `/nginx-ingress-controller` in the Nginx ingress deployment without it the pods crashloopBackOff. Is this expected?
![Screen Shot 2022-06-30 at 12 49 29 PM](https://user-images.githubusercontent.com/87133789/176734213-7d58b4ad-1ad7-41b6-a638-731695bb63cb.png)

**Second Question**
The nginx config map looks as below and it is passed as an arg with the correct namespace and name in the deployment definition. However, no changes are taking effect or is getting ignored. For ex: keepalive-timeout: 90s is getting ignored as the value for this parameter at /etc/nginx/nginx.conf is displayed as 65s. Same with the SSL protocols config, log-format.

```
kind: ConfigMap
apiVersion: v1
metadata:
  name: nginx-config
  namespace: default
data:
  brotli-level: '4'
  brotli-types: >-
    text/css text/* text/javascript message/* application/x-javascript
    application/json application/xml application/atom+xml application/xaml+xml
  enable-brotli: 'true'
  enable-modsecurity: 'true'
  fail-timeout: 15s
  http-snippet: |
    server {
      listen 18080;

      location /nginx_status {
        allow all;
        stub_status on;
      }

      location / {
        return 404;
      }
    }
  keepalive-timeout: 90s
  log-format: >-
    $remote_addr - $request_id - [$proxy_add_x_forwarded_for] - $remote_user
    [$time_local] "$request" $status $body_bytes_sent "$http_referer"
    "$http_user_agent" $request_length $request_time [$proxy_upstream_name]
    $upstream_addr $upstream_response_length $upstream_response_time
    $upstream_status
  proxy-body-size: 100m
  proxy-connect-timeout: 100s
  proxy-read-timeout: 100s
  proxy-send-timeout: 100s
  ssl-protocols: TLSv1.2
  use-forwarded-headers: 'true'
  use-http2: 'true'
  worker-connections: '10240'
```

The argument for the  configmap in the deployment is as shown in the below screenshot
![Screen Shot 2022-06-30 at 12 57 46 PM](https://user-images.githubusercontent.com/87133789/176735119-d45ebfa4-4caf-4c41-adbc-4b4120201957.png)

Any help on this much appreciated!!

**Comment by jasonwilliams14** [comment]
@rrachitha 
Are you using our Dockerfile to build the image? Or are you building your own customer NGINX Ingress controller image.

would it be possible to share your deployment file so I can review?
Looking at your snippet in the above, the arguments do not look correct.
Reviewing your deployment file will help me identify the issue.
Thx

**Comment by jjngx** [comment]
No answer for 3 years, closing.
